src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -138,24 +150,60 @@ public boolean canUseLocalCache() {
 
         return false;
     }
-
+        
     /**
-     * App needs to give permission to AccountManager to use broker.
+     * To verify if App gives permissions to AccountManager to use broker.
+     * 
+     * @throws UsageAuthenticationException
+     * @return boolean If the GET_ACCOUNTS, MANAGE_ACCOUNTS, USE_CREDENTIALS permissions are granted in the Manifest.xml
      */
-    private boolean verifyManifestPermissions() {
-        PackageManager pm = mContext.getPackageManager();
-        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                "android.permission.GET_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.USE_CREDENTIALS", mContext.getPackageName());
-        if (!permission) {
-            Logger.w(TAG, "Broker related permissions are missing for GET_ACCOUNTS, MANAGE_ACCOUNTS, USE_CREDENTIALS",
+    private boolean verifyManifestPermissions() throws UsageAuthenticationException {
+        final String methodName = ":verifyManifestPermissions";
+        final PackageManager packageManager = mContext.getPackageManager();
+        List<String> permerssionMissing = new ArrayList<String>();
```

Comment:
You can use StringBuffer is you just want to append all the missing permissions.